### PR TITLE
Allow partial rustfmt.toml

### LINKED ISSUE #255: Partial rustfmt.toml not supported
Currently one has to specify all configs in the `rustfmt.toml` even if all we want is to only change a single config (e.g. `struct_trailing_comma`). It should just use defaults for the ones that are not provided by the `rustfmt.toml`

### PR DESCRIPTION
**Pull request by cburgdorf**
With this change one can use a config file
that only specifies a subset of config keys
to overwrite. E.g. a config file that looks like this

```
struct_trailing_comma = "Never"
struct_lit_trailing_comma = "Never"
```

Fixes #255

### CODE REVIEW

File: src/config.rs [hunk 1]
```diff
@@ -63,7 +78,9 @@ macro_rules! create_config {
                         println!("\n\nParsed:\n{:?}", parsed);
                         panic!();
                     }
-                }
+                };
+                Config::default()
+                       .fill_from_parsed_config(&parsed_config)
```
Comment (nrc) [inline src/config.rs]: nit: this will all fit on one line

File: src/config.rs [hunk 2]
```diff
@@ -52,10 +52,25 @@ macro_rules! create_config {
...
+        pub struct ParsedConfig {
+            $(pub $i: Option<$ty>),+
+        }
+
         impl Config {
+
+          fn fill_from_parsed_config(mut self, parsed: &ParsedConfig) -> Config {
+          $(
+              if let Some(val) = parsed.$i {
+                self.$i = val;
```
Comment (nrc) [inline src/config.rs]: nit: please use 4 space indent here

File: src/config.rs [hunk 3]
```diff
@@ -52,10 +52,25 @@ macro_rules! create_config {
...
         }
 
+        #[derive(RustcDecodable, Clone)]
+        pub struct ParsedConfig {
+            $(pub $i: Option<$ty>),+
+        }
+
         impl Config {
+
+          fn fill_from_parsed_config(mut self, parsed: &ParsedConfig) -> Config {
```
Comment (nrc) [inline src/config.rs]: and here

File: src/config.rs [hunk 4]
```diff
@@ -52,10 +52,25 @@ macro_rules! create_config {
             $(pub $i: $ty),+
         }
 
+        #[derive(RustcDecodable, Clone)]
+        pub struct ParsedConfig {
```
Comment (nrc) [inline src/config.rs]: Could you add a comment please explaining what this struct is for and how the parsing the process works

**Comment by nrc** [comment]
Thanks for the PR, it looks good and I'd like to have this functionality. There are a few formatting nits to fix and it could do with a comment. r+ with that.

File: src/config.rs (hunk 2 above)
Comment (cburgdorf) [inline src/config.rs]: yeah, this is atom not allowing per project configs and me switching constantly between different projects...fix is under way ;)

**Comment by cburgdorf** [comment]
Thanks for the feedback. I updated the PR.

**Comment by nrc** [comment]
Awesome. thanks!

**Comment by cburgdorf** [comment]
Super cool! That was actually the first time that I ever wrote macro code :)

**Pull request merged**